### PR TITLE
ref(source-amazon-ads): migrate stream `SponsoredDisplayReportStream` to amazon ads reports V3

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolve https://github.com/airbytehq/airbyte-internal-issues/issues/10360

---

First step in amazon ads enhancement, see full tasklist in https://github.com/airbytehq/airbyte-internal-issues/issues/10429

---

## How

refactor stream: `SponsoredDisplayReportStream` to use reports v3
Remove streams:
- SponsoredBrandsVideoReportStream (v2)
- SponsoredBrandsReportStream (v2)

## Review guide

1. `airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/display_report.py`

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/display_report.py
Comment (bazarnov) [inline airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/display_report.py:582]: Can we have a mapping of values {reportType: (reportTypeId, groupBy, filters)} instead of if...else, and parse it accordingly using get_body() method or similar?

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/display_report.py
Comment (artem1205) [reply airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/display_report.py:582]: We _can_ do this, but to be honest, I don't want to waste time on this, since we [plan](https://github.com/airbytehq/airbyte-internal-issues/issues/10429) to migrate this connector to Async Low-code

File: airbyte-integrations/connectors/source-amazon-ads/metadata.yaml
Comment (maxi297) [inline airbyte-integrations/connectors/source-amazon-ads/metadata.yaml:37]: As a user, I would cope better with the issue (especially with the small delay we ask for this change) if we were to indicate that this is because of an API deprecation. Something like: "The Report API v2 is getting deprecated in favor of V3. Given this change, fields available through the `sponsoredDisplayReportStream` stream will change."

File: airbyte-integrations/connectors/source-amazon-ads/metadata.yaml
Comment (maxi297) [inline airbyte-integrations/connectors/source-amazon-ads/metadata.yaml:42]: I would also assume `SponsoredBrandsReportStream` would change based on the description of the ticket. Is there something I'm missing here?

**Review by maxi297 — COMMENTED** [review]
I've added a couple of comments before we merge this. Can we also resolve the merge conflicts?

File: airbyte-integrations/connectors/source-amazon-ads/metadata.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-amazon-ads/metadata.yaml]: agreed, updated

File: airbyte-integrations/connectors/source-amazon-ads/metadata.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-amazon-ads/metadata.yaml:44]: updated

**Review by maxi297 — APPROVED** [review]
LGTM! Can we get a review from @katmarkham for the breaking change?

File: airbyte-integrations/connectors/source-amazon-ads/metadata.yaml
Comment (nataliekwong) [inline airbyte-integrations/connectors/source-amazon-ads/metadata.yaml:38]:
```suggestion
          "The Report API V2 is being deprecated in favor of V3. Given this change, fields available through the sponsoredDisplayReportStream stream will change,
```

File: airbyte-integrations/connectors/source-amazon-ads/metadata.yaml
Comment (nataliekwong) [inline airbyte-integrations/connectors/source-amazon-ads/metadata.yaml:39]:
```suggestion
          and streams `SponsoredBrandsReportStream` `SponsoredBrandsVideoReportStream` will become unavailable. We recommend using `SponsoredBrandsV3ReportStream` as an alternative stream with the same dataset."
```

**Review by katmarkham — APPROVED** [review]

**Comment by katmarkham** [comment]
thanks for your updates @maxi297 !

File: docs/integrations/sources/amazon-ads-migrations.md
Comment (nataliekwong) [inline docs/integrations/sources/amazon-ads-migrations.md:6]:
```suggestion
The `SponsoredDisplayReportStream` stream now has an updated schema, thanks to a recent change in the Amazon Ads API. You can find more details in the [Amazon Migration Guide (metrics)](https://advertising.amazon.com/API/docs/en-us/reference/migration-guides/reporting-v2-v3#metrics).

Please note that SponsoredBrandsReportStream and SponsoredBrandsVideoReportStream will become unavailable as a result of the deprecation of API V2. We recommend switching to SponsoredBrandsV3ReportStream as a great alternative.
```

**Pull request merged**
